community-tutorials/assign-additional-ip-to-lxc-container/01-en.md

```diff
@@ -0,0 +1,470 @@
+config:
+  security.nesting: true
+  user.network-config: |
+    version: 2
+    ethernets:
+        eth0:
+            addresses:
+            - 37.121.182.6/32
+            nameservers:
+                addresses:
+                - 8.8.8.8
+                search: []
+            routes:
+            -   to: 0.0.0.0/0
+                via: 169.254.0.1
+                on-link: true
+description: Routed LXD profile
+devices:
+  eth0:
+    ipv4.address: 37.121.182.6
+    nictype: routed
+    parent: eth0
+    host_name: veth-bbb
+    type: nic
+name: bbb
+used_by:
+```
+
+The configuration **`security.nesting: true`** is needed in order to
+run docker inside the container. However if the container is
+unprivileged it does not really have any security implications.
+
+I am using the second public IP that I deleted from the host interface
+(`37.121.182.6/32`). For this reason this profile can be used only for
+one container. To build other containers like this, we should make a
+copy of the profile and modify it.
+
+The **`config.user.network-config`** part is about the configuration
+of the container (through cloud-init). The gateway is `169.254.0.1`.
+
+Notice that **`devices.eth0.nictype`** is **`routed`**. I could have
+used an **ipvlan** type as well, and most of the configurations would
+be almost the same, however it seems that in this case the container
+cannot ping the public IP of the host, and I don't want this.
+
+The field **`devices.eth0.host_name`** sets the name of the virtual
+interface that will be created on the host. If it is not specified,
+then a random name will be used each time the container is
+started. But this would make difficult the specification of the
+firewall rules (that we will see later).
+
+The field **`devices.eth0.parent`** is the name of the interface on
+the host where this virtual interface will be attached. In my case
+this is not really necessary and can be left out or commented.
+
+# Step 4 - Launching the container
+
+The latest stable version of BBB requires `ubuntu:18.04`.
+
+```bash
+lxc launch ubuntu:18.04 bbb --profile default --profile bbb
+lxc list
+lxc list -c ns4t
+lxc info bbb
+```
+
+With the command `ip addr` you can notice that a new interface named
+`veth-bbb` has been created, and it has the IP `169.254.0.1/32`.
+
+With the command `ip ro` you can notice that a route like this has
+been added:
+
+```
+37.121.182.6 dev veth-bbb scope link
+```
+
+Try also these commands:
+
+```bash
+lxc exec bbb -- ip addr
+lxc exec bbb -- ip ro
+```
+
+Notice that the interface inside the container has IP
+`37.121.182.6/32` and the default gateway is `169.254.0.1`.
+
+We can also ping from the host to `37.121.182.6`, however from the
+container we cannot ping to the host or outside (to the Internet):
+
+```bash
+ping 37.121.182.6
+lxc exec bbb -- ping 169.254.0.1
+lxc exec bbb -- ping 8.8.8.8
+```
+
+# Step 5 - Fix networking
+
+The problem is that I have installed `firewalld` on the host and it
```

Comment:
If you are using any other packages than explicity installed in this tutorial, please add these installations also to the tutorial. Simply add `apt install firewalld` before the others firewall-cmd commands

Reply:
You are right, I should mention the installation of `firewalld`, maybe in the introduction.

Reply:
I added some instructions about installing `firewalld` and `fail2ban` at the requirements.